cmd/ack-generate/command/olm.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+
+	generate "github.com/aws-controllers-k8s/code-generator/pkg/generate"
+	ackgenerate "github.com/aws-controllers-k8s/code-generator/pkg/generate/ack"
+	olmgenerate "github.com/aws-controllers-k8s/code-generator/pkg/generate/olm"
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/spf13/cobra"
+)
+
+const (
+	olmConfigFileSuffix = "olmconfig.yaml"
+)
+
+// optOLMConfigPath
+var optOLMConfigPath string
+var optDisableCommonLinks bool
+var optDisableCommonKeywords bool
+
+// olmCmd is the command that generates a service ClusterServiceVersion base
+// for generating an operator lifecycle manager bundle.
+var olmCmd = &cobra.Command{
+	Use: "olm <service> <version>",
+	Short: `Generate an Operator Lifecycle Manager's ClusterServiceVersion
+resources for AWS service API. Expects a configuration file to be passed in
+via the --olm-config option, or otherwise reads the config at the root of the
+project with the filename <servicealias>-olmconfig.yaml`,
```

Comment:
Yes, definitely! I haven't tested running `ack-generate` from a directory outside of the `code-generator` repository, so the "default" behavior is to look for the file at the base of `code-generator` if nothing is passed in via the `--olm-config` flag. In practice, the olm-config file is in a weird place because it's not generated and always needs to be read from some external source. It's similar to the API definitions that are pulled in from aws-sdk-go in my mind. On the other hand, it makes logical sense to store alongside the service controller.

As an aside - In theory, I could have looked for it in the default location for output files (which I believe is `$svcAlias/`, but the file is _never_ generated, so it didn't make sense to me to look for it at that location. 

TL;DR - Yes! The plan was to include it in service controller repositories. The minimal configuration, in theory, should mean it should need infrequent updates.